### PR TITLE
Fix bug in `layer.save_own_variables`

### PR DESCRIPTION
**Pull request by Hilly12**
`layer.save_own_variables(store)` doesn't save sub-layer variables. Can change to `self.trainable_variables + self.non_trainable_variables` if we don't want to save seed generator state.

Repro:

```
model = keras.Sequential([
    keras.layers.Conv2D(32, kernel_size=(3, 3), activation="relu"),
    keras.layers.MaxPooling2D(pool_size=(2, 2)),
    keras.layers.Conv2D(64, kernel_size=(3, 3), activation="relu"),
    keras.layers.MaxPooling2D(pool_size=(2, 2)),
    keras.layers.Flatten(),
    keras.layers.Dense(256, activation="relu"),
    keras.layers.Dense(10),
])
model.compile(
    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
    optimizer=keras.optimizers.SGD(0.1),
    metrics=[keras.metrics.SparseCategoricalAccuracy()],
)
model.build((None, 28, 28, 1))

>>> print(model.variables)
[<KerasVariable shape=(3, 3, 1, 32), dtype=float32, path=sequential/conv2d/kernel>,
 <KerasVariable shape=(32,), dtype=float32, path=sequential/conv2d/bias>,
 <KerasVariable shape=(3, 3, 32, 64), dtype=float32, path=sequential/conv2d_1/kernel>,
 <KerasVariable shape=(64,), dtype=float32, path=sequential/conv2d_1/bias>,
 <KerasVariable shape=(1600, 256), dtype=float32, path=sequential/dense/kernel>,
 <KerasVariable shape=(256,), dtype=float32, path=sequential/dense/bias>,
 <KerasVariable shape=(256, 10), dtype=float32, path=sequential/dense_1/kernel>,
 <KerasVariable shape=(10,), dtype=float32, path=sequential/dense_1/bias>]

store = {}
model.save_own_variables(store)

>>> print(store)
{}
```

### CODE REVIEW

**Comment by fchollet** [comment]
That's the intended behavior. That's why there's "own" in the method name.

The way you save all variables non-redundantly is by recursively traversing the trackable structure and calling `save_own_variables` on the leaves.

Note that `save_own_variable` is never intended to be called by users. It's intended to be *implemented* by layer creators who need to customize how saving works for their layer (note: needing this is exceedingly rare).

If you just want to save the variables of a layer, do `.save_weights()`.

**Comment by Hilly12** [comment]
My bad, turns out the issue I was running into was due to an inconsistency in how the model was constructed in the place where it was saved versus where it was loaded.
